trade-finance-manager-ui/server/controllers/case/activity/index.test.js

```diff
@@ -71,3 +78,212 @@ describe('GET activity', () => {
+  const normalComment = 'test';
+
+  beforeEach(() => {
+    api.getDeal = () => Promise.resolve(mockDeal);
+  });
+
+  it('should return to comment box if longer than 1000 characters', async () => {
+    const req = {
+      params: {
+        _id: mockDeal._id,
+      },
+      body: {
+        comment: longComment,
+      },
+      session,
+    };
+
+    await activityController.postComment(req, res);
+    expect(res.render).toHaveBeenCalledWith('case/activity/activity-comment.njk',
+      {
+        dealId: mockDeal.dealSnapshot._id,
+        user: session.user,
+        maxCommentLength: 1000,
+      });
+  });
+  it('should return render activities page with all-activity filter after posting comment', async () => {
+    const req = {
+      params: {
+        _id: mockDeal._id,
+      },
+      body: {
+        comment: normalComment,
+      },
+      session,
+    };
+    const activities = activityController.mappedActivities(mockDeal.tfm.activities);
+
+    await activityController.postComment(req, res);
+    expect(res.render).toHaveBeenCalledWith('case/activity/activity.njk',
+      {
+        activePrimaryNavigation: 'manage work',
+        activeSubNavigation: 'activity',
+        deal: mockDeal.dealSnapshot,
+        tfm: mockDeal.tfm,
+        dealId: mockDeal.dealSnapshot._id, // eslint-disable-line no-underscore-dangle
+        user: session.user,
+        selectedActivityFilter: CONSTANTS.ACTIVITIES.ACTIVITY_FILTERS.ALL,
+        activities,
+      });
+  });
+  it('should return render activities page with all-activity filter after not posting a comment', async () => {
+    const req = {
+      params: {
+        _id: mockDeal._id,
+      },
+      body: {
+        comment: '',
+      },
+      session,
+    };
+    const activities = activityController.mappedActivities(mockDeal.tfm.activities);
+
+    await activityController.postComment(req, res);
+    expect(res.render).toHaveBeenCalledWith('case/activity/activity.njk',
+      {
+        activePrimaryNavigation: 'manage work',
+        activeSubNavigation: 'activity',
+        deal: mockDeal.dealSnapshot,
+        tfm: mockDeal.tfm,
+        dealId: mockDeal.dealSnapshot._id, // eslint-disable-line no-underscore-dangle
+        user: session.user,
+        selectedActivityFilter: CONSTANTS.ACTIVITIES.ACTIVITY_FILTERS.ALL,
+        activities,
+      });
+  });
+});
```

Comment:
Can't see any tests for the `/not-found` redirect. Not sure if this is missing in GEF generally. Not a blocker

Reply:
I can do it for each one.  Thought that the first one  was tested has it so didn't need to further down?

Reply:
If you look at test coverage report in the "lines covered" column or "missing", for this controller, you'll probably see the lines where "redirect to not-found" are, so i'd try to make sure it's all covered.

Reply:
This is what I mean about the test coverage @Zainzzkk  (i'm improving now)
![Screenshot 2021-11-24 at 11 30 20](https://user-images.githubusercontent.com/2463648/143230507-9ec215ff-9ac1-420c-8f94-f05a2f0ab4ee.png)